crud/room_utils.py

```diff
@@ -0,0 +1,257 @@
+    """Remove existing room."""
+    _room = get_room(db=db, room_id=room_id)
+    if not _room:
+        raise HTTPException(
+            status_code=404, detail=f"No room found with id {room_id}"
+        )
+    db.delete(_room)
+    db.commit()
+    return f"Deleted room with id {room_id}"
+
+
+def get_room_booking_status(db: Session, room_id: int):
+    """Get booking state of specific room."""
+    _room = get_room(db=db, room_id=room_id)
+    if not _room:
+        raise HTTPException(
+            status_code=404, detail=f"No room found with id {room_id}"
+        )
+    return _room.booking_status
+
+
+def get_room_cleanliness_status(db: Session, room_id: int):
+    """Get cleanliness state of specific room."""
+    _room = get_room(db=db, room_id=room_id)
+    if not _room:
+        raise HTTPException(
+            status_code=404, detail=f"No room found with id {room_id}"
+        )
+    return _room.cleanliness_status
+
+
+# Room types CRUD
+
+
+def get_room_types(db: Session, skip: int = 0, limit: int = 0):
+    """Get all room types."""
+    return db.query(RoomType).offset(skip).limit(limit).all()
+
+
+def get_room_type(db: Session, room_type_id: int):
+    """Get room type by id."""
+    _room_type = db.query(RoomType).filter(RoomType.id == room_type_id).first()
+    if not _room_type:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    return _room_type
+
+
+def create_room_type(db: Session, room_type: RoomTypeCreate):
+    """Create new room type."""
+    _room_type = RoomType(
+        name=room_type.name, capacity=room_type.capacity, price=room_type.price
+    )
+    db.add(_room_type)
+    db.commit()
+    db.refresh(_room_type)
+    return _room_type
+
+
+def update_room_type(
+    db: Session, room_type_id: int, room_type: RoomTypeCreate
+):
+    """Update an existing room type."""
+    _room_type = get_room_type(db=db, room_type_id=room_type_id)
+    if not _room_type:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    _room_type.name = room_type.name
+    _room_type.capacity = room_type.capacity
+    _room_type.price = room_type.price
+    db.commit()
+    db.refresh(_room_type)
+    return _room_type
+
+
+def delete_room_type(db: Session, room_type_id: int):
+    """Remove an existing room type."""
+    _room_type = get_room_type(db=db, room_type_id=room_type_id)
+    if not _room_type:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    db.delete(_room_type)
+    db.commit()
+    return _room_type
+
+
+# Room features CRUD
+
+
+def get_features(db: Session, skip: int = 0, limit: int = 100):
+    """Get all features."""
+    return db.query(Feature).offset(skip).limit(limit).all()
+
+
+def get_feature(db: Session, feature_id: int):
```

Comment:
if you return all feature properties on get_features this function looks a bit redundant